### PR TITLE
[SYCLomatic] Add test for `cuCtxPushCurrent`, `cuCtxPopCurrent`, `cuStreamQuery`, `cuCtxEnablePeerAccess` and `cuDeviceCanAccessPeer`

### PR DESCRIPTION
**Pull request by the-slow-one**
As described in the title adding end-to-end test for the mentioned API.

### CODE REVIEW

File: features/feature_case/peer_access_using_driver_api/peer_access_using_driver_api.cu [hunk 1]
```diff
@@ -0,0 +1,32 @@
+#include <cuda.h>
+#include <iostream>
```
Comment (zhimingwang36) [inline features/feature_case/peer_access_using_driver_api/peer_access_using_driver_api.cu:10]: add a license header for the test files.

File: features/feature_case/driverStreamAndEvent/driverStreamQuery.cu [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#include <cuda.h>
+#include <iostream>
```
Comment (zhimingwang36) [inline features/feature_case/driverStreamAndEvent/driverStreamQuery.cu:10]: how this test is triggerred to run?

File: features/feature_case/peer_access_using_driver_api/peer_access_using_driver_api.cu (hunk 1 above)
Comment (zhimingwang36) [inline features/feature_case/peer_access_using_driver_api/peer_access_using_driver_api.cu:10]: how the test is triggered to run?

File: features/feature_case/driverStreamAndEvent/driverStreamQuery.cu (hunk 2 above)
Comment (the-slow-one) [reply features/feature_case/driverStreamAndEvent/driverStreamQuery.cu:10]: `driverStreamAndEvent` is already a part of global variable `exec_tests`. The test file is added to _TEMPLATE_driverStreamEvent_ now.

File: features/feature_case/peer_access_using_driver_api/peer_access_using_driver_api.cu (hunk 1 above)
Comment (the-slow-one) [reply features/feature_case/peer_access_using_driver_api/peer_access_using_driver_api.cu:10]: Done

Comment (the-slow-one) [reply features/feature_case/peer_access_using_driver_api/peer_access_using_driver_api.cu:10]: Registerd the test now. See changes in _features.xml_ in same PR.

**Pull request merged**